### PR TITLE
Fix Wan feature cache CFG status toggle

### LINKED ISSUE #902: [Bug] WanTransformerInferFirstBlock lost the class method 'switch_status'
Seems 8689e4c7b4838992a38f882030af29af5a60372c causing class method switch_status of class WanTransformerInferFirstBlock lost, which is  add in commit dcaefe635a3b8da6a1f8f80f4fd06d556b16bf50 
the diff file of this two commit is as follow,

with this bug, wan model cannot work properly using FBCache, I assume.  

Can you help confirm it? thx @helloyongyang 

```txt


commit 8689e4c7b4838992a38f882030af29af5a60372c
Author: helloyongyang <yongyang1030@163.com>
Date:   Fri Aug 15 11:52:00 2025 +0000

    update tea cache

diff --git a/lightx2v/models/networks/wan/infer/transformer_infer.py b/lightx2v/models/networks/wan/infer/transformer_infer.py
index f53842a4..e0ab522a 100755
--- a/lightx2v/models/networks/wan/infer/transformer_infer.py
+++ b/lightx2v/models/networks/wan/infer/transformer_infer.py
@@ -78,11 +78,6 @@ class WanTransformerInfer(BaseTransformerInfer):
         else:
             self.infer_func = self._infer_without_offload
 
-        self.infer_conditional = True
-
-    def switch_status(self):
-        self.infer_conditional = not self.infer_conditional
-
     def _calculate_q_k_len(self, q, k_lens):
         q_lens = torch.tensor([q.size(0)], dtype=torch.int32, device=q.device)
         cu_seqlens_q = torch.cat([q_lens.new_zeros([1]), q_lens]).cumsum(0, dtype=torch.int32)

commit dcaefe635a3b8da6a1f8f80f4fd06d556b16bf50
Author: Yang Yong(雍洋) <yongyang1030@163.com>
Date:   Sun Jun 29 21:41:27 2025 +0800

    update feature caching (#78)
    
    Co-authored-by: Linboyan-trc <1584340372@qq.com>

diff --git a/lightx2v/models/networks/wan/infer/transformer_infer.py b/lightx2v/models/networks/wan/infer/transformer_infer.py
index 421e3534..174be2e6 100755
--- a/lightx2v/models/networks/wan/infer/transformer_infer.py
+++ b/lightx2v/models/networks/wan/infer/transformer_infer.py
@@ -4,10 +4,11 @@ from lightx2v.common.offload.manager import (
     WeightAsyncStreamManager,
     LazyWeightAsyncStreamManager,
 )
+from lightx2v.common.transformer_infer.transformer_infer import BaseTransformerInfer
 from lightx2v.utils.envs import *
 
 
-class WanTransformerInfer:
+class WanTransformerInfer(BaseTransformerInfer):
     def __init__(self, config):
         self.config = config
         self.task = config["task"]
@@ -49,8 +50,10 @@ class WanTransformerInfer:
         else:
             self.infer_func = self._infer_without_offload
 
-    def set_scheduler(self, scheduler):
-        self.scheduler = scheduler
+        self.infer_conditional = True
+
+    def switch_status(self):
+        self.infer_conditional = not self.infer_conditional
 
     def _calculate_q_k_len(self, q, k_lens):
         q_lens = torch.tensor([q.size(0)], dtype=torch.int32, device=q.device)

```

### PR DESCRIPTION
**Pull request by Fatemanx**
## Summary
This PR fixes the Wan feature-caching CFG state handling by using `scheduler.infer_condition` as the single source of truth.

`WanModel._infer_cond_uncond()` already sets `self.scheduler.infer_condition` before each conditional/unconditional inference pass. The feature-caching transformer now reads that scheduler state directly instead of keeping a separate `infer_conditional` flag.

## Changes
- remove the separate Wan transformer `infer_conditional` / `switch_status()` state
- update Wan feature-caching branches to read `self.scheduler.infer_condition` directly
- replace the old switch-status regression test with a scheduler-state regression test for `WanTransformerInferFirstBlock`

## Validation
- reproduced the original missing-state issue before the first fix
- verified the updated regression test with:
  ```bash
  HOME=/tmp XDG_CACHE_HOME=/tmp SKIP_PLATFORM_CHECK=1 \
  conda run -n exp_env python -B -m unittest test_cases.test_wan_feature_cache_cfg_state
  ```
- verified there are no remaining `infer_conditional` / `switch_status` references under Wan code
- ran `pre-commit run --files` on the touched files

## Notes
This PR only changes the CFG state source used by Wan feature caching. It does not change `BaseTransformerInfer` or `scheduler.infer_condition` setup.

Closes #902

### CODE REVIEW

File: lightx2v/models/networks/wan/infer/transformer_infer.py [hunk 1]
```diff
@@ -72,13 +72,17 @@ def rope_wrapper(xq, xk, cos_sin_cache):
             self.enable_head_parallel = False
             self.seq_p_tensor_fusion = False
         self.infer_func = self.infer_without_offload
+        self.infer_conditional = True
```
Comment (Musisoul) [inline lightx2v/models/networks/wan/infer/transformer_infer.py]: The current code already tracks the CFG state via `scheduler.infer_condition` in [`model.py`](https://github.com/ModelTC/LightX2V/blob/main/lightx2v/models/networks/wan/model.py#L105), and that value changes with conditional/unconditional CFG inference.

So it would be better to use `self.scheduler.infer_condition` directly instead of introducing a separate `self.infer_conditional` state. That would also let us remove the `switch_status()` logic and avoid the risk of the two states getting out of sync.

Comment (Fatemanx) [reply lightx2v/models/networks/wan/infer/transformer_infer.py]: Thanks for the review. I updated the PR to use `self.scheduler.infer_condition` directly as the single source of truth for Wan feature-caching CFG state, and removed the separate `infer_conditional` / `switch_status()` state to avoid divergence.

**Review by Musisoul — APPROVED** [review]
